.github/workflows/nodejs.yml

```
@@ -0,0 +1,30 @@
+name: Node CI
+
+on: [push]
```

Instead of running on every push, I think it'd be better to be explicit. How about pushes to master and pull requests against master? Please double-check this is valid YAML before accepting the change haha, I may have messed up tabbing.

```suggestion
on:
   push:
      branches:
         - master
         
   pull_request:
      branches:
         - master
```

---

What's the benefit of not running on all commits? IMO realizing early on that something broke is useful, and the limit of free actions should not be an issue for this amount of CI.

Of course, if you think it's better for this repo to only run on master interaction, this change looks fine.

---

Can you be sure this will ensure actions runs on opened pull requests if you don't include the `pull_request` item?

---

Yes, they will run on the final commit thats included in the PR 👌🏻
The only time this will  skip commits is if multiple are part of a single push. It will only run on the latest commit of a push.

Concerning the tests: If I understand GitHubs secret injection correctly, they will only run once a commit is not pushed to a fork, but to this repository. Meaning they might not run in PRs until merged 🤔
*Maybe* adding the PR event will trigger the tests as well, but I don't think so. Not injecting the secrets in forks has a main reason: One could just add a CI that publishes the secrets. Therefore, running it in PRs that have not been merged doesn't make sense, as they pose the same risk.

---

Makes sense to me, nice!